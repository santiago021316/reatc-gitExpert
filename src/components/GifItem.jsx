import React from 'react'

export const GifItem = ({title,url,id}) => {

  return (
    <div className='card img'>
        <img src={url} alt={title}  />
        <p>{title}</p>
    </div>
   
  )
}
