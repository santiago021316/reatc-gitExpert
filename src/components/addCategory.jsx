import { useState } from "react"
import  PropTypes   from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

const [inputValue, setInputValue] = useState('')

          const onInputChange = (event)=>{setInputValue(event.target.value)}
      
          const onSubmit = (event) => {
            //  console.log('submit desde react testing')
                event.preventDefault();
             if(inputValue.trim().length <= 1) return;

                setInputValue('')
                onNewCategory(inputValue.trim())
          }

 return (
    <form onSubmit= {onSubmit} aria-label="form">
        <input type="text"
               placeholder="buscar gif"  
               value={inputValue}
               onChange={onInputChange}
                />
        </form>
         )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}

