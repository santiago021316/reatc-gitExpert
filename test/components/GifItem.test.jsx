import { render ,screen} from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";





describe('pruebas en el componente  gifItem', () => { 
    const title = 'saitana'
    const url = 'https://saitana.org/jpg'


test('debe hacer match con el snapshot', () => { 
    const {container} = render(<GifItem title={title} url={url}/>)
    expect(container).toMatchSnapshot()
    
})

test('debe mostarr la imagen con el url indicado', () => { 

    render(<GifItem title={title} url={url}/>);
    
    const {src, alt} = screen.getByRole('img')
    //  expect(screen.getByRole('img').src).toBe(url)
    //  expect(screen.getByRole('img').src).toBe(url)
     
    expect(src).toBe(url)
    expect(alt).toBe(title)
 })

 test('debe de mostrar el titulo en el componenete', () => { 
    render(<GifItem title={title} url={url}/>);
    expect(screen.getByAltText(title)).toBeTruthy()
 })

 })