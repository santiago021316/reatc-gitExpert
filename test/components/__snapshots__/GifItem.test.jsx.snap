// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente  gifItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card img"
  >
    <img
      alt="saitana"
      src="https://saitana.org/jpg"
    />
    <p>
      saitana
    </p>
  </div>
</div>
`;
